### PR TITLE
AEROGEAR-1970 Run ansible scripts inside AWS EC2 instance

### PR DESCRIPTION
**Pull request by vchepeli**
## Motivation
[AEROGEAR-1970](https://issues.jboss.org/browse/AEROGEAR-1970)
## Description
[AEROGEAR-1970](https://issues.jboss.org/browse/AEROGEAR-1970)

## Progress
- [x] Finished task
- [] TODO

## Additional Notes

### CODE REVIEW

**Comment by maleck13** [comment]
@AdamSaleh Looking

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+node("mobile-core-install-slave") {
+    stage("Checkout") {
+      checkout scm
+    }
+
+    stage("Run Ansible scripts") {
+        def metadata_endpoint = "http://169.254.169.254/latest/meta-data"
```
Comment (maleck13) [inline Jenkinsfile]: does this have to be hardcoded? do we want a follow up issue to remove it or will it never change

**Comment by maleck13** [comment]
@aidenkeating ok to take a quick look at this with your knowledge of the installer

File: Jenkinsfile (hunk 1 above)
Comment (vchepeli) [reply Jenkinsfile]: Yes this should be hardcoded. Thats static variable

**Comment by david-martin** [comment]
@vchepeli Technically, this is good to merge.
However, I would like to see some example in the readme/docs for the command & args to use to provision to a remote instance.

**Comment by vchepeli** [comment]
@david-martin they are in [Jenkinsfile](https://github.com/aerogear/mobile-core/pull/39/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R20). Arguments should be same, there is only AWS specific public hostname and routing suffix

**Comment by david-martin** [comment]
Thanks for the clarification @vchepeli.
Is this intended just for automation/testing, or is it intended for someone evaluating mobile in a remote openshift cluster?

File: installer/roles/ansible-service-broker-setup/files/provision-ansible-service-broker.sh [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 set -e
 
-readonly DOCKERHUB_USER="${1}"
-readonly DOCKERHUB_PASS="${2}"
-readonly DOCKERHUB_ORG="${3}"
-readonly LAUNCH_APB_ON_BIND="${4}"
-readonly TAG="${5}"
-readonly WILDCARD_DNS="${6}"
-readonly ANSIBLE_SERVICE_BROKER_NAMESPACE="${7}"
+readonly DOCKERHUB_USER=${1:?"[ERROR]You must provide a dockerhub username."}
```
Comment (david-martin) [inline installer/roles/ansible-service-broker-setup/files/provision-ansible-service-broker.sh]: @vchepeli I'm hitting an issue with local provsion with this change.

My dockerhub password has a symbol `)` in it, which doesn't get parsed OK.
```
"stderr_lines": ["/bin/sh: -c: line 0: syntax error near unexpected token `)'"
```

Can we add back quotes around this?

**Comment by AdamSaleh** [comment]
@david-martin this is mostly for automation/testing.

File: installer/roles/ansible-service-broker-setup/files/provision-ansible-service-broker.sh (hunk 2 above)
Comment (vchepeli) [reply installer/roles/ansible-service-broker-setup/files/provision-ansible-service-broker.sh]: Added already. @david-martin can you try it again please?

File: installer/roles/ansible-service-broker-setup/tasks/main.yml [hunk 3]
```diff
@@ -58,7 +58,7 @@
       mode: u+x
 
   - name: Execute OpenShift Ansible Broker (OAB) provision script
-    shell: bash -e /tmp/provision-ansible-service-broker.sh "{{ dockerhub_username }}" "{{ dockerhub_password }}" "{{ dockerhub_org }}" "{{ launch_apb_on_bind }}" "{{dockerhub_tag}}" "{{ wildcard_dns_host }}" "{{ ansible_service_broker_namespace }}"
+    shell: "bash -e /tmp/provision-ansible-service-broker.sh {{ dockerhub_username }} {{ dockerhub_password }} {{ dockerhub_org }} {{ launch_apb_on_bind }} {{dockerhub_tag}} {{ cluster_public_ip }} {{ wildcard_dns_host }} {{ ansible_service_broker_namespace }}"
```
Comment (david-martin) [inline installer/roles/ansible-service-broker-setup/tasks/main.yml]: @vchepeli looks like it needs to be escaped here too. Not sure why the escaping was removed

Comment (vchepeli) [reply installer/roles/ansible-service-broker-setup/tasks/main.yml]: > Not sure why the escaping was removed

There was no comment why it escapes every symbol and I removed it

**Comment by david-martin** [comment]
Thanks @vchepeli :+1:

**Pull request merged**